Remove last of GLOB_BRACE

iCaspar opened the pull request:
Closes #146 

This PR is to remove the last 3 instances of GLOB_BRACE from `rocket_clean_home()`
Also adds unit/integration tests for the function.

iCaspar committed:
Remove GLOB_BRACE where no braces

iCaspar committed:
Refactor glob to regex

iCaspar committed:
Revert "Refactor glob to regex"

This reverts commit 6c8ad45a4553f060210b301038bafcbcce97a6f7.

iCaspar committed:
Revert "Refactor glob to regex"

This reverts commit 6c8ad45a4553f060210b301038bafcbcce97a6f7.

# Conflicts:
#	inc/functions/files.php

iCaspar committed:
Add escape for `/` in regex

crystinutzaa commented on inc/functions/files.php:
```diff
@@ -613,10 +613,13 @@ function rocket_clean_home( $lang = '' ) {
 	do_action( 'before_rocket_clean_home', $root, $lang ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
 
 	// Delete homepage.
-	$files = glob( $root . '/{index,index-*}.{html,html_gzip}', GLOB_BRACE | GLOB_NOSORT );
```
Will this delete also other files and folders? I did not tested it, but I am concerned it might delete other files except index files.

iCaspar replied:
The new regex should not delete any other files. @piotrbak has given it some initial tests (and found in the process this separate issue #3444) and confirmed that it should be ok, but has not had full QA yet.
Some discussion came up last week re: glob use in general. See my comment [here](https://github.com/wp-media/wp-rocket/issues/146#issuecomment-753994031) 
@crystinutzaa tagging you for Code Review.

iCaspar committed:
Merge branch 'develop' into fix/GLOB_BRACE-remove-last

iCaspar committed:
Merge branch 'develop' into fix/GLOB_BRACE-remove-last

iCaspar committed:
Add integration test for rocket_clean_home()

iCaspar committed:
Clean up regex

iCaspar commented:
Looks like we have some integration test breakage around the PDFEmbedder ☹️ 
But I've added a test here and cleaned up the regex a bit.
Ready for review again.
We should definitely merge this and get rid of the issue; the other non-critical glob ideas are "nice to have" possibilities for the future.

iCaspar committed:
Add cases for mobile and cookie variations

Mai-Saad reviewed (commented):
Working as expected, failing tests either need test maintenance or it's an issue on the trunk
[testrail-report-307.pdf](https://github.com/wp-media/wp-rocket/files/7557877/testrail-report-307.pdf)
Notes:
- Assuming that update theme only clears the wp-rocket cache 
- Switching theme clear wp-rocket and /min/1 cache

iCaspar committed:
Merge branch 'develop' into fix/GLOB_BRACE-remove-last
